### PR TITLE
Improve lon diff

### PR DESCRIPTION
**Pull request by jedwards4b**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Avoid erroneous CESM error messages by insuring lon_diff works correctly.
- [X] Developer(s): 
    Jim Edwards
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Tested using SMS_Ld2.ne30pg3_t232.BLT1850_v0c.cheyenne_intel.allactive-defaultio
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:
We need to assure that the result of both mod calculations are 0<= x < 360, we do that by adding 720 to both values before computing the mod base 360.   For this test case the number of (incorrect) error messages in the CESM log went from 87726 to 0.

### CODE REVIEW

File: cicecore/drivers/nuopc/cmeps/ice_mesh_mod.F90 [hunk 1]
```diff
@@ -637,13 +637,10 @@ subroutine ice_mesh_check(gcomp, ice_mesh, rc)
              lon(n) = tlon(i,j,iblk)*rad_to_deg
              lat(n) = tlat(i,j,iblk)*rad_to_deg
 
-             tmplon = lon(n)
-             if(tmplon < c0)tmplon = tmplon + c360
-
              ! error check differences between internally generated lons and those read in
-             diff_lon = abs(mod(lonMesh(n) - tmplon,360.0))
+             diff_lon = abs(mod(2*c360+lonMesh(n),c360) - mod(2*c360+lon(n),c360))
```
Comment (DeniseWorthen) [inline cicecore/drivers/nuopc/cmeps/ice_mesh_mod.F90:641]: I have a vague memory that UFS had an issue w/ how this check was originally coded and the tmplon may be the result of that? I will make a quick test on your modification.

Comment (jedwards4b) [reply cicecore/drivers/nuopc/cmeps/ice_mesh_mod.F90:641]: Thanks!

**Comment by dabail10** [comment]
I am putting this directly into the Consortium.

**Comment by jedwards4b** [comment]
Can you point me to that PR so that I can follow it?  Thanks

**Comment by dabail10** [comment]
Still not sure why this works. When I compute this in ncl:

print(mod(720+359.9,360))
359.9
print(mod(720+0.0001,360))
0.0001

So, these are still different. It seems to work in the Fortran code though. We need a negative mod essentially.

**Pull request closed without merging**